source/fundamentals/data-formats/codecs.txt

```diff
@@ -11,38 +11,337 @@ Codecs
+The ``Codec`` abstraction allows you to map types and fields directly to
+your domain objects rather than to an intermediate map-based one such
+as ``Document`` or ``BsonDocument``.
+
+We describe the following interfaces and classes and provide examples of
+how you can specify custom encoding and decoding logic using them in the
+following sections:
+
+- :ref:`Codec <codecs-codec>`
+- :ref:`CodecRegistry <codecs-codecregistry>`
+- :ref:`CodecProvider <codecs-codecprovider>`
+- :ref:`Custom Codec Example <codecs-custom-example>`
+
+If you are customizing encoding and decoding logic for Plain old Java objects
+(POJOs), read our guide on :doc:`POJO Customization </fundamentals/data-formats/pojo-customization>`.
+
+.. _codecs-codec:
+
+Codec
+-----
+
+The ``Codec`` interface contains abstract methods for serializing and
+deserializing Java objects to BSON data. You can define your conversion logic
+between BSON and your Java object in your implementation of this interface.
+
+To implement the ``Codec`` interface, override the ``encode()``, ``decode()``,
+and ``getEncoderClass()`` abstract methods.
+
+The ``encode()`` method requires the following parameters:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 20
+
+   * - Parameter Type
+     - Description
+
+   * - ``writer``
+     - An instance of a class that implements ``BsonWriter``, an interface type
+       that exposes methods for writing a BSON document. For example, the
+       ``BsonBinaryWriter`` implementation writes to a binary stream of data.
+       Use this instance to write your BSON value using the appropriate
+       write method.
+
+   * - ``value``
+     - The data that your implementation encodes. The type must match the type
+       variable assigned to your implementation.
+
+   * - ``encoderContext``
+     -  Contains meta information about the Java object data that it encodes
+        to BSON including whether to store the current value in a
+        MongoDB collection.
+
+The ``decode()`` method returns your Java object instance populated with the
+value from the BSON data. This method requires the following parameters:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 20
+
+   * - Parameter Type
+     - Description
+
+   * - ``bsonReader``
+     - An instance of a class that implements ``BsonReader``, an interface type
+       that exposes methods for reading a BSON document. For example, the
+       ``BsonBinaryReader`` implementation reads from a binary stream of data.
+
+   * - ``decoderContext``
+     - Contains information about the BSON data that it decodes to a Java
+       object.
+
+The ``getEncoderClass()`` method returns a class instance of the Java class
+since Java cannot infer the type due to due to type erasure.
+
+.. _codecs-powerstatus-codec:
+
+See the following ``PowerStatusCodec`` class that implements ``Codec`` in
+order to convert the Java ``String`` values "on" and "off" to
```

Comment:
Are we switching to the convention  where we use monospace and language syntax for values, so`"on"` and `"off"`?

Reply:
Not that I know of, but maybe you have a different convention in mind? 

I've mentioned what convention I've been following for all the docs repos that our team currently works on, but can explain again. 

I'd like to stick to that for at least the current documentation repos, but if there's a pressing need to change it, we should discuss as a team and apply the convention across the docs.

Reply:
Sounds good. You are correct that we did not reach resolution [on point 4 of the Writing Hints and Tips for Drivers](https://docs.google.com/document/d/1SxlDvx4MYKBE6-LYGp5i5uuHfY1ZUYYlhc6syEdta5M/edit). I'll try and restart conversation on the doc.

Marking comment as resolved as no changes are requested.